fix: Decrease default admission controller timeout

mikebryant opened the pull request:
### What does this PR do?

Decreases the default timeout set for the MutatingWebhookConfiguration to a value less than the [default client timeout](https://github.com/kubernetes/client-go/blob/f6ce18ae578c8cca64d14ab9687824d9e1305a67/transport/cache.go#L93)

### Motivation

We've been trying to track down some unexplained errors in our GKE cluster, of the following form: `Error creating: Timeout: request did not complete within requested timeout`.
This is from the CronJob controller, relating to the creation of Pods in the apiserver.

We found https://github.com/kubernetes/kubernetes/issues/71508, which talks about how the default client timeout is 30s, so if the webhooks have a timeout equal or greater, the client can give up before the apiserver has, which means the `failurePolicy: ignore` has no effect - it doesn't happen soon enough.

### Describe how to test your changes

We've set      
```
clusterAgent:
  env:
  - name: DD_ADMISSION_CONTROLLER_TIMEOUT_SECONDS
     value: "10"
```
in our helm installation, and things have been working correctly.

mikebryant committed:
fix: Decrease default admission controller timeout

If the timeout is >= the default client timeout of 30s, this can lead to the client believing the request failed. failurePolicy: Ignore still needs the api server to wait the full timeout before responding to requests

See https://github.com/kubernetes/kubernetes/issues/71508 for more details

ahmed-mez commented on pkg/config/config.go:
```diff
@@ -718,7 +718,7 @@ func InitConfig(config Config) {
 	config.BindEnvAndSetDefault("admission_controller.mutate_unlabelled", false)
 	config.BindEnvAndSetDefault("admission_controller.port", 8000)
-	config.BindEnvAndSetDefault("admission_controller.timeout_seconds", 30) // 30s corresponds to the default value set by the Kubernetes API
+	config.BindEnvAndSetDefault("admission_controller.timeout_seconds", 10) // 30s corresponds to the default value set by the Kubernetes API
```
```suggestion
	config.BindEnvAndSetDefault("admission_controller.timeout_seconds", 10) // in seconds (see kubernetes/kubernetes#71508)
```

ahmed-mez reviewed (commented):
Hi @mikebryant ! 

Thank you for the fix, this makes sense!

Could you just add a release note in the `releasenotes-dca/notes` folder? The process is documented [here](https://github.com/DataDog/datadog-agent/blob/master/docs/dev/contributing.md#reno).

Side note: Even though decreasing the timeout is needed to avoid the problem described in the issue you linked, we highly recommend investigating why the admission requests were timing out. Was/were the Datadog Cluster Agent pod(s) not ready during these timeouts?
We'd be happy to help you move the investigation forward and fix the root cause, don't hesitate to [contact our support team](https://www.datadoghq.com/support/)!

Thanks!

mikebryant committed:
Fix comment

Co-authored-by: Ahmed Mezghani <38987709+ahmed-mez@users.noreply.github.com>

mikebryant committed:
docs: Add release note

mikebryant committed:
chore: Move note to correct place

mikebryant commented:
Added a release note :)

We had some issues on the cluster at the time due to some incorrectly configured Node Pools. I don't have full visibility on what was happening at the time, the problem cleared up after we removed the broken pool. Mostly just trying to cover all the bases here :)

ahmed-mez reviewed (commented):
Thanks! Glad to hear that the problem was solved! 

Two small comments about the release note then we can merge this :)

mikebryant committed:
Update releasenotes-dca/notes/fix:-Decrease-default-admission-controller-timeout-ac077793f18af85f.yaml

Co-authored-by: Ahmed Mezghani <38987709+ahmed-mez@users.noreply.github.com>

mikebryant committed:
Make file name shorter
